src/util/check.h

```diff
@@ -42,10 +42,6 @@ class NonFatalCheckError : public std::runtime_error
         }                                                         \
     } while (false)
 
-#if defined(NDEBUG)
-#error "Cannot compile without assertions!"
-#endif
```

Comment:
This is where our `Assert` is defined, so I think it makes sense to have the compiler check here.